### PR TITLE
Configure for readthedocs

### PR DESCRIPTION
**Pull request by icemac**
The result looks like this: https://pytest-rerunfailures--284.org.readthedocs.build/284/

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the OS, Python version and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.12"
```
Comment (hugovk) [inline .readthedocs.yaml:11]: Can we use newer?
```suggestion
  os: ubuntu-24.04
  tools:
    python: "3.13"
```
Or latest?
```suggestion
  os: ubuntu-lts-latest
  tools:
    python: "3"
```

**Comment by webknjaz** [comment]
I recommend wiring Sphinx via tox to avoid maintaining two integrations.

Here's how you call it then: https://github.com/ansible/awx-plugins/blob/e613d5a/.readthedocs.yaml.

**Comment by icemac** [comment]
@webknjaz This looks way more complicated than what we currently have.

**Comment by webknjaz** [comment]
> @webknjaz This looks way more complicated than what we currently have.

It's necessary if we don't want to have the workflow of building locally for contributors to be "go read the source code and attempt to guess which command to run, where, in what env and how".

**Comment by icemac** [comment]
@webknjaz I got your point and I added tox environment to build the docs locally. But I am not going to use it at readthedocs. Using their workflow seems way easier.

**Comment by icemac** [comment]
I do not plan to improve this PR any further.
So I enabled auto-merge and started waiting for a review.
If someone wants to adopt this PR for further improvements, I can disable auto-merge again.

**Review by hugovk — APPROVED** [review]
Thanks!

**Pull request merged**